src/app-layout/utils/use-observed-element.ts

```diff
@@ -1,12 +1,13 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 import { useResizeObserver } from '@cloudscape-design/component-toolkit/internal';
-import { useCallback, useState } from 'react';
+import React, { useCallback, useState } from 'react';
 
-export function useObservedElement(selector: string) {
+export function useObservedElement(rootElementRef: React.RefObject<HTMLElement>, selector: string) {
   const getElement = useCallback(() => {
+    const document = rootElementRef.current?.ownerDocument ?? window.document;
```

Comment:
Needed to change this to run the new dev pages without any unsafe CSP. Maybe it will fix some real use-cases too.